providers/azure/client/api/api.go

```diff
@@ -56,12 +61,20 @@ func CheckResponse(res *http.Response) error {
 			}
 			jerr.Error.Body = string(slurp)
 			jerr.Error.URL = res.Request.URL.String()
-			return jerr.Error
+			e = jerr.Error
+		}
+	} else {
+		e = &Error{
+			StatusCode: res.StatusCode,
+			Body:       string(slurp),
+			Header:     res.Header,
 		}
 	}
-	return &Error{
-		StatusCode: res.StatusCode,
-		Body:       string(slurp),
-		Header:     res.Header,
+
+	switch e.StatusCode {
```

Comment:
wondering whether we should implement it in the aci provider like what you did for the other providers.

Reply:
Yeah, I definitely cheated here.

Reply:
Moved this handling into the provider.